pkg/asset/cluster/metadata.go

```diff
@@ -0,0 +1,109 @@
+package cluster
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/cluster/aws"
+	"github.com/openshift/installer/pkg/asset/cluster/libvirt"
+	"github.com/openshift/installer/pkg/asset/cluster/openstack"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/pkg/errors"
+)
+
+const (
+	metadataFileName = "metadata.json"
+)
+
+// Metadata contains information needed to destroy clusters.
+type Metadata struct {
+	file *asset.File
+}
+
+var _ asset.WritableAsset = (*Metadata)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (m *Metadata) Name() string {
+	return "Metadata"
+}
+
+// Dependencies returns the direct dependencies for the metadata
+// asset.
+func (m *Metadata) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.ClusterID{},
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the metadata asset.
+func (m *Metadata) Generate(parents asset.Parents) (err error) {
+	clusterID := &installconfig.ClusterID{}
+	installConfig := &installconfig.InstallConfig{}
+	parents.Get(clusterID, installConfig)
+
+	if installConfig.Config.Platform.None != nil {
```

Comment:
With these changes, `metadata.json` has become information and not just a file used by the `destroy` command. It would rather see consistency where the file is generated even when the platform is `None`. With that said, I can live with the changes as they are, and we can address what to do about the `None` platform for later releases.